src/scripting/modules/bittorrent/torrent_info_wrapper.cpp

```diff
@@ -13,25 +14,36 @@ duk_ret_t torrent_info_wrapper::construct(duk_context* ctx)
     int t = duk_get_type(ctx, 0);
     libtorrent::torrent_info* info;
 
-    if (t == DUK_TYPE_STRING)
+    try
     {
-        std::string file(duk_require_string(ctx, 0));
-        // TODO: error handling
-        info = new libtorrent::torrent_info(file);
+        if (t == DUK_TYPE_STRING)
+        {
+            std::string file(duk_require_string(ctx, 0));
+            info = new libtorrent::torrent_info(file);
+        }
+        else if (t == DUK_TYPE_BUFFER)
+        {
+            duk_size_t size;
+            const char* buffer = static_cast<const char*>(duk_require_buffer(ctx, 0, &size));
+            info = new libtorrent::torrent_info(buffer, size);
+        }
+
+        duk_push_this(ctx);
+        common::set_pointer<libtorrent::torrent_info>(ctx, -2, info);
+
+        duk_push_c_function(ctx, finalize, 1);
+        duk_set_finalizer(ctx, -2);
     }
-    else if (t == DUK_TYPE_BUFFER)
+    catch (const libtorrent::libtorrent_exception& ex)
     {
-        duk_size_t size;
-        const char* buffer = static_cast<const char*>(duk_require_buffer(ctx, 0, &size));
-        // TODO: error handling
-        info = new libtorrent::torrent_info(buffer, size);
+        BOOST_LOG_TRIVIAL(warning) << "Invalid torrent file: " << ex.what();
+        return DUK_RET_UNSUPPORTED_ERROR;

```

Comment:
There doesn't seem to be a more-specific error code to return here, nor does the webui show any failure information. Returning an error here does get an error message printed from RPC.js however, yielding the second of these two messages:

```
(0000002e) 01:16:52.699148: <warning> Invalid torrent file: expected value (list, dict, int or string) in bencoded string
(0000002f) 01:16:52.714772: <error> Error when executing RPC method 'webui.addTorrent': Error: unsupported error (rc -51)
```